### PR TITLE
Feature/add point endpoint

### PR DESCRIPTION
**Pull request by vincentsarago**
continues #146 

The mai change is that I'm introducing `multi_points_pgstac` which replace rio-tiler's `multi_values` method (because we use item dict instead of url, so it's not hashable as dict key). 

In `multi_points_pgstac` I'm constructing an Item ID using the id and collection which we says are `mandatory` (this is not asserted but should)

@hrodmn could you have a look and tell me your tough

### CODE REVIEW

File: titiler/pgstac/factory.py [hunk 1]
```diff
@@ -901,6 +902,53 @@ def feature_image(
...
+                        **dataset_params,
+                        **pgstac_params,
+                    )
+
+            return {
+                "coordinates": [lon, lat],
+                "values": [
+                    (src, pts.data.tolist(), pts.band_names) for src, pts in values
+                ],
+            }
```
Comment (vincentsarago) [inline titiler/pgstac/factory.py:950]: This is now equivalent to https://github.com/developmentseed/titiler/blob/main/src/titiler/mosaic/titiler/mosaic/factory.py#L672-L677 and remove the need of custom model

File: titiler/pgstac/mosaic.py [hunk 2]
```diff
@@ -33,6 +33,30 @@
...
+
+    Custom version of `rio_tiler.task.multi_values` which
+    use constructed `item_id` as dict key.
+
+    """
+    tasks = create_tasks(reader, asset_list, threads, *args, **kwargs)
+
+    out: Dict[str, Any] = {}
+    for val, asset in filter_tasks(tasks, allowed_exceptions=allowed_exceptions):
+        item_id = f"{asset['collection']}/{asset['id']}"
```
Comment (vincentsarago) [inline titiler/pgstac/mosaic.py:54]: collection and id are mandatory

**Review by hrodmn — COMMENTED** [review]
@vincentsarago that looks like a cleaner version of what I attempted in #146 - thank you for fixing it up!

**Comment by hrodmn** [comment]
> @hrodmn I'm having second though on the endpoint path `/{lon},{lat}/values`
> 
> I know we already have `/{lon},{lat}/assets` which might be the reason for the choice but at the same time in titiler (single item) we use `/point/{lon},{lat}`
> 
> would you be ok if we change this to `/point/{lon},{lat}` ?

That's fine with me. When I started working on it I went with `/point/{lon},{lat}` but changed my mind after finding `/{lon},{lat}/assets`. I think it makes sense to stay consistent with `titiler`.

**Pull request merged**